pkg/controller/statusmanager/status_manager.go

```diff
@@ -103,11 +104,11 @@ func (status *StatusManager) set(reachedAvailableLevel bool, conditions ...opv1a
 
 	// In case that the status field has been updated with "Progressing" condition, make sure that
 	// "Ready" condition is set to False, even when not explicitly set.
-	progressingCondition := getCondition(&config.Status, opv1alpha1.NetworkAddonsConditionProgressing)
+	progressingCondition := conditionsv1.FindStatusCondition(config.Status.Conditions, conditionsv1.ConditionProgressing)
```

Comment:
you could simplify this to `conditionsv1.IsConditionTrue(config.Status.Conditions, conditionsv1.ConditionProgressing)`.

Reply:
Thanks David, will fix it. I think we are still having issues during upgrade. Plus I will need to implement Upgradable in a following PR, but should not be much hassle.